smithy-model/src/test/resources/software/amazon/smithy/model/shapes/idl-serialization/cases/service-mixins.smithy

```diff
@@ -0,0 +1,39 @@
+$version: "2.0"
+
+namespace smithy.example
+
+service MixedService with MixinService {
+}
+
+@internal
+@mixin
+service MixinService {
+    version: "2021-12-31"
+    operations: [
+        MixinOperation
+    ]
+    resources: [
+        MixinResource
+    ]
+    errors: [
+        MixinError
+    ]
+    rename: {
```

Comment:
Could the shape mixing this in extend this rename map? Can you add a test if so?